src/App.js

```diff
@@ -11,8 +11,37 @@ import {
   Schedule
 } from './pages'
 import Page from './components/Page'
+import { db } from './utility/firebase'
+import { DB_COLLECTION, DB_HACKATHON } from './utility/Constants'
+import notifications from './utility/notifications'
+
+// only notify user if announcement was created within last 5 secs
+const notifyUser = (announcement) => {
+  const isRecent = new Date() - new Date(announcement.timestamp) < 5000
+  if (isRecent && notifications.areEnabled()) {
+    notifications.trigger("New Announcement", announcement.content)
+  }
+}
 
 function App() {
+  useEffect(() => {
+    const unsubscribe = db
+      .collection(DB_COLLECTION)
+      .doc(DB_HACKATHON)
+      .collection('Announcements')
+      .orderBy('timestamp', 'desc')
+      .onSnapshot(querySnapshot => {
+        // firebase doc that triggered db change event
+        const changedDoc = querySnapshot.docChanges()[0]
+
+        // don't want to notify on 'remove' + 'modified' db events
```

Comment:
maybe we want to notify on 'modified' events tho 🤷‍♂️

Reply:
okok I know this is technically out of scope, but thoughts on adding a 'isModified' prop to the `Announcement` component and add a little `edited @ HH:MM:SS` message so people know what's been updated